### PR TITLE
[ci skip] CodeTriage badge

### PR DESCRIPTION
**Pull request by schneems**
[Code Triage](https://www.codetriage.com/) is an app I've maintained for the past 4-5 years with the intent of getting people involved in open source. It sends subscribers a random open issue for them to help "triage". For Ruby projects such as "rails/rails" you can also subscribe to documentation. For example you can get a few random documented methods, or if you want to write docs, get undocumented methods.

The initial approach was inspired by seeing the work of the small core team spending countless hours asking "what rails version was this in" and "can you give us an example app". The idea is to outsource these small interactions to a huge team of volunteers and let the core team focus on their work.

The purpose of the badge is to give more people an easier way to start contributing to Rails. Here's what it currently looks like:

[Code Triage Badge](https://www.codetriage.com/rails/rails)

The number is how many people are currently subscribed (a.k.a. "helpers") to the project on CodeTriage, the color is based off of the number of open issues in the project. You can see an example of this badge on another popular open source repo [Crystal](github.com/crystal-lang/crystal/).

> For context to non-rails core: I also maintain sprockets (though a release hasn't happened in some time, sorry), and I have commit to Rails. I'm not some rando trying to push arbitrary links to READMEs on GitHub.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ Everyone interacting in Rails and its sub-projects' codebases, issue trackers, c
 
 ## Code Status
 
-[![Build Status](https://travis-ci.org/rails/rails.svg?branch=master)](https://travis-ci.org/rails/rails)
+[![Build Status](https://travis-ci.org/rails/rails.svg?branch=master)](https://travis-ci.org/rails/rails) [![Code Triage Badge](https://www.codetriage.com/rails/rails/badges/users.svg)](https://www.codetriage.com/rails/rails)
```
Comment (sgrif) [inline README.md]: Should this go under the `Contributors` header instead?

Comment (schneems) [inline README.md]: Great idea. I was on auto pilot and just grouped it with other badges. Will move it up

Comment (schneems) [inline README.md]: Updated

**Comment by matthewd** [comment]
I'm really not comfortable endorsing / sending people to CT while it encourages people to document things we have no interest in documenting. It creates enough of a "thanks but no thanks" flow already, but for us to reject changes that are recommended by something we've pointed them to seems quite rude indeed.

**Comment by schneems** [comment]
We should be nodoc-ing things that don't need to be documented. Or otherwise indicate that it doesn't have docs on purpose. FWIW I think even private methods can benefit from docs as long as they're also marked as private. There are some internal active record bits that are really hard to develop with, I wrote docs for them and have to reference them all the time (when I'm working with internal active record bits).

Im not sure how many people use the documentation feature, I can pull some numbers though.

**Comment by matthewd** [comment]
Documenting nontrivial internals is indeed useful. But by volume, most private methods are trivial helpers, whose name adequately describes what they do.

The first link I just saw on the list was https://github.com/rails/rails/blob/9914964514fc43f6eadba06a47fccdf95b7806f7/activerecord/lib/active_record/associations/singular_association.rb/#L33. Short of rehashing the behaviour of the entire scoping system on every such method, there's really not much to add. Especially not in a subclass.

The fourth link was https://github.com/rails/rails/blob/9914964514fc43f6eadba06a47fccdf95b7806f7/activemodel/lib/active_model/type/value.rb/#L89, which **is nodoc**. :confused:

Better big-to-medium picture docs on AR internals sound like a great idea, but I think they belong on classes at least, if not as a separate "mini-guide".

---

More importantly, IMO, we have a bunch of real genuine public API methods that aren't documented. A lot of them are again going to be of the "the class + method names say it all" variety, but adding a redundant one-sentence description could still have some value there. And an unfamiliar volunteer surely has a far better chance of working out what a public method does than something deep in the bowels [to a degree that it's useful to record their understanding as official documentation].

**Comment by schneems** [comment]
wanted to mention an idea I had. Find an hour or two where several contributors could be available for live questions about documentating Rails code. And ready to merge in doc commits. Could be good to help steer people towards more high quality documentation actions. Think of it as a doc-a-palooza.

Regarding the :nodoc: I need to skip those. It was in a "read" doc and not a "write" doc section though, right?

I agree with the comment about class level docs with examples being the most useful. I envisioned people incidentally finding undocumented classes as they looked through undocumented methods.

**Pull request merged**